torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -1106,6 +1108,153 @@ def arg_extractor(combine_fn, input, dim):
         )
 
 
+class ScanHigherOrderVariable(TorchHigherOrderOperatorVariable):
+    @raise_hard_error_if_graph_break(
+        reason="scan must be captured completely with torch.compile."
+    )
+    def call_function(
+        self,
+        tx: "InstructionTranslator",
+        args: List[VariableTracker],
+        kwargs: Dict[str, VariableTracker],
+    ) -> VariableTracker:
+        from .builder import SourcelessBuilder, wrap_fx_proxy
+
+        args, kwargs = LazyVariableTracker.realize_all((args, kwargs))
+
+        def arg_extractor(combine_fn, input, init, dim):
+            return combine_fn, input, init, dim
+
+        combine_fn, input, init, dim = arg_extractor(*args, **kwargs)
+
+        if input.python_type() != list:
+            unimplemented(
+                f"Expected input to be a list of tensors but got {input.python_type()}",
+            )
+        assert isinstance(input, torch._dynamo.variables.lists.BaseListVariable)
+        if init.python_type() != list:
+            unimplemented(
+                f"Expected input to be a list of tensors but got {input.python_type()}",
+            )
+        assert isinstance(init, torch._dynamo.variables.lists.BaseListVariable)
+
+        # Trace the subgraph
+        # TODO: Fix these pointless new_empty calls appearing in the dynamo output graph.
+        sub_args = [
+            leaf.call_method(
+                tx,
+                "new_empty",
+                args=(
+                    SourcelessBuilder.create(
+                        tx,
+                        leaf_init.size
```

Comment:
why they're called leaf? It would be better to keep consistent with `elems` `inits` or `xs`, "init" in jax style across all places.

Reply:
I tried to rename everything. Now it is `input` (to be consistent with the associative_scan) and `init` as it is called in jax.lax.scan